GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -1094,14 +1099,25 @@ private void UpdatePlaceholders()
 
                 this.blobSizes.Flush();
 
-                // Waiting for the UpdateOrDeleteFilePlaceholder to fill the folderPlaceholdersToKeep before trying to remove folder placeholders
-                // so that we don't try to delete a folder placeholder that has file placeholders and just fails
-                foreach (PlaceholderListDatabase.PlaceholderData folderPlaceholder in placeholderListCopy.Where(x => x.IsFolder).OrderByDescending(x => x.Path))
-                {
-                    this.TryRemoveFolderPlaceholder(folderPlaceholder, updatedPlaceholderList, folderPlaceholdersToKeep);
+                using (BlobSizes.BlobSizesConnection blobSizesConnection = this.blobSizes.CreateConnection())
+                {
+                    // Waiting for the UpdateOrDeleteFilePlaceholder to fill the folderPlaceholdersToKeep before trying to remove folder placeholders
```

Comment:
Where are we waiting? I don't see it in the code.

Reply:
I see now that you just moved this comment, so it's not new. But I'm having a hard time parsing what it means.

Reply:
The wait happens in `ProcessListOnThreads`:

```
for (int i = 0; i < processThreads.Length; i++)
{
    processThreads[i].Join();
}
```
But I that's not clear from the comment.  I can change it to something like:

```
// Don't attempt to delete folder placeholders until UpdateOrDeleteFilePlaceholder 
// has finished populating folderPlaceholdersToKeep
```

Reply:
I think I would just remove the comment. My guess is that it was written before the ProcessListOnThreads was factored out, and we were actually waiting on threads to complete right before this comment. Now the comment is just confusing - of course we're waiting for it, this code won't run until that other method returns, and folderPlaceholdersToKeep is an output of that method.